flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/ApplicationReconciler.java

```diff
@@ -145,6 +147,15 @@ public void deploy(
         var relatedResource = ctx.getResource();
         var status = relatedResource.getStatus();
         var flinkService = ctx.getFlinkService();
+
+        if (spec.getJob().getState().equals(JobState.SUSPENDED)
+                || status.getReconciliationStatus()
+                        .getState()
+                        .equals(ReconciliationState.UPGRADING)) {
```

Comment:
Do we need any of these checks? If we are deploying a job then the previous health info should probably be cleared always

Reply:
Well, such case we're more on the safe side so not needed. Removed.